tools/nblIntrinicGenerator/Readme.md

```diff
@@ -0,0 +1,212 @@
+# Intrinsic generator tool for hlsl
+
+This tool generates hlsl code that exposes spirv instructions.
+
+It was made as manual writing and making sure these instructions are up to date proved to be too time consuming.
+
+To run the app use 
+```
+python main.py [grammar] [fields] [type mapping] [output] [-v|--verbose]
+```
+* grammar - Filepath to a JSON file containing the instructions you want to expose. For example, this is a valid grammar file `include/spirv/unified1/spirv.core.grammar.json`.
+
+* fields - Filepath to a JSON file containing a list of builtin variables and functions to expose in the output file.
+
+* type mapping - Filepath to a JSON file containg information about types of parameters and return types of exposed functions.
+
+* output - where to write the generated HLSL file
+
+* verbose - print additional information, as well as any warnings such as omitted fields 
+
+Or run example that takes in no arguments
+```
+python runExample.py
+```
+
+
+
+# Type Map JSON
+
+An example JSON file that contains the information about spirv function types might look like this:
+```
+{
+    "default": "uint32_t",
+    "classMappings":[
+        {
+            "class": "Atomic",
+            "types": [
+                {
+                    "kind": "IdResult",
+                    "type": "T"
+                },
+                {
+                    "kind": "IdRef",
+                    "name": "Pointer",
+                    "type": "P",
+                    "attributes": [
+                        "vk::ext_reference"
+                    ]
+                },
+                {
+                    "kind": "IdRef",
+                    "name": "Value",
+                    "type": "T"
+                },
+                {
+                    "kind": "IdRef",
+                    "name": "Comparator",
+                    "type": "T",
+                    "const": false
+                }
+            ]
+        },
+        {
+            "class": "Conversion",
+            "types": [
+                {
+                    "kind": "IdResult",
+                    "type": "T"
+                },
+                {
+                    "kind": "IdRef",
+                    "name": "Operand",
+                    "type": "U"
+                }
+            ]
+        }
+        
+    ],
+    "instructionMappings":[
+        {
+            "instruction": "GLSLstd450MatrixInverse",
+            "types": [
+                {
+                    "kind": "IdResult",
+                    "type": "T"
+                }
+            ]
+        }
+    ]
+}
+```
+
+Default type declared in type mapping JSON file must not be generic, preferably primitive. `uint32_t` is recommended.
```

Comment:
huh?

Reply:
Just a handicap from me, as i did not see the reason to pull any fancy schenenigans with default type